devdoc/interlocked_requirements.md

```diff
@@ -0,0 +1,298 @@
+```
+
+**SRS_INTERLOCKED_43_004 [** `interlocked_and_64` shall perform an atomic AND operation on the specified 64-bit integer values.**]**
+
+## interlocked_and_8
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_and_8, volatile int8_t*, destination, int8_t, value);
+```
+
+**SRS_INTERLOCKED_43_005 [** `interlocked_and_8` shall perform an atomic AND operation on the specified 8-bit integer values.**]**
+
+## interlocked_compare_exchange
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_compare_exchange, volatile int32_t*, destination, int32_t, exchange, int32_t, comperand);
+```
+
+**SRS_INTERLOCKED_43_006 [** `interlocked_compare_exchange` shall perform an atomic compare-and-exchange operation on the specified values. The function compares two specified 32-bit values and exchanges with another 32-bit value based on the outcome of the comparison.**]**
+
+## interlocked_compare_exchange_128
+
+```c
+MOCKABLE_FUNCTION(, bool, interlocked_compare_exchange_128, volatile int64_t*, destination, int64_t, exchange_low, int64_t, exchange_high, int64_t, comperand_result);
+```
+
+**SRS_INTERLOCKED_43_007 [** `interlocked_compare_exchange_128` shall perform an atomic compare-and-exchange operation on the specified values. The function compares two specified 128-bit values and exchanges with another 128-bit value based on the outcome of the comparison.**]**
+
+## interlocked_and_16
+
+```c
+MOCKABLE_FUNCTION(, int16_t, interlocked_and_16, volatile int16_t*, destination, int16_t, value);
+```
+
+**SRS_INTERLOCKED_43_008 [** `interlocked_and_16` shall perform an atomic compare-and-exchange operation on the specified values. The function compares two specified 16-bit values and exchanges with another 16-bit value based on the outcome of the comparison.**]**
+
+## interlocked_compare_exchange_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_compare_exchange_64, volatile int64_t*, destination, int64_t, exchange, int64_t, comperand);
+```
+
+**SRS_INTERLOCKED_43_009 [** `interlocked_compare_exchange_64` shall perform an atomic compare-and-exchange operation on the specified values. The function compares two specified 64-bit values and exchanges with another 64-bit value based on the outcome of the comparison.**]**
+
+## interlocked_compare_exchange_pointer
+
+```c
+MOCKABLE_FUNCTION(, void*, interlocked_compare_exchange_pointer, volatile void*, destination, void*, exchange, void*, comperand);
+```
+
+**SRS_INTERLOCKED_43_010 [** `interlocked_compare_exchange_pointer` shall perform an atomic compare-and-exchange operation on the specified values. The function compares two specified pointer values and exchanges with another pointer value based on the outcome of the comparison.**]**
+
+## interlocked_decrement
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_decrement, volatile int32_t*, addend);
+```
+
+**SRS_INTERLOCKED_43_011[** `interlocked_decrement` shall decrement (decrease by one) the value of the specified 32-bit variable as an atomic operation.**]**
```

Comment:
>the specified 32-bit variable [](start = 99, length = 29)

I would write it as `*addend`. 

Then I don't get into "is it a variable or parameter"? and it is extremely crisp and clear specified because the name is in the function declaration.